### PR TITLE
Remove atom references

### LINKED ISSUE #2400: Remove remaining references to ATOM

### PR DESCRIPTION
**Pull request by ElizabethOkerio**
### Issues

*This pull request fixes #2400 *

### Description

This PR removes references to atom references. Have left out these references in tests because most of the atom tests are commented out and in some instances the logic is not called anywhere. I'd like cleaning this in the tests clean-up work. 

### Checklist (Uncheck if it is not completed)

- [ ] *Test cases added*
- [ ] *Build and test with one-click build and test script passed*

### Additional work necessary

*If documentation update is needed, please add "Docs Needed" label to the issue and provide details about the required document change in the issue.*

### CODE REVIEW

File: src/Microsoft.OData.Client/Materialization/CollectionValueMaterializationPolicy.cs [hunk 1]
```diff
@@ -97,13 +97,13 @@ internal object CreateCollectionInstance(IEdmCollectionTypeReference edmCollecti
...
             Debug.Assert(clientCollectionType != null, "clientCollectionType!=null");
-            return CreateCollectionInstance(edmCollectionTypeReference, clientCollectionType, () => DSClient.Strings.AtomMaterializer_MaterializationTypeError(clientCollectionType.FullName));
+            return CreateCollectionInstance(edmCollectionTypeReference, clientCollectionType, () => DSClient.Strings.Materializer_MaterializationTypeError(clientCollectionType.FullName));
         }
 
         /// <summary>
         /// Applies collectionValue item to the provided <paramref name="collectionInstance"/>.
         /// </summary>
-        /// <param name="collectionProperty">Atom property containing materialized Collection items.</param>
+        /// <param name="collectionProperty">Property containing materialized Collection items.</param>
```
Comment (gathogojr) [inline src/Microsoft.OData.Client/Materialization/CollectionValueMaterializationPolicy.cs:106]: nit:
```suggestion
        /// <param name="collectionProperty">Property containing materialized collection items.</param>
```

File: src/Microsoft.OData.Client/Materialization/ODataEntriesEntityMaterializer.cs [hunk 2]
```diff
@@ -86,7 +86,7 @@ internal override long CountValue
         {
             get
             {
-                throw new InvalidOperationException(DSClient.Strings.MaterializeFromAtom_CountNotPresent);
+                throw new InvalidOperationException(DSClient.Strings.MaterializeFromObject_CountNotPresent);
```
Comment (gathogojr) [inline src/Microsoft.OData.Client/Materialization/ODataEntriesEntityMaterializer.cs:89]: nit: Maybe "MaterializeFromEntry" better than "MaterializeFromObject" given that you're dropping "atom" from "atom entry" in other places...

File: src/Microsoft.OData.Client/MaterializeFromAtom.cs [hunk 3]
```diff
@@ -21,7 +21,7 @@ namespace Microsoft.OData.Client
     #endregion Namespaces
 
     /// <summary>
-    /// Use this class to materialize objects from an application/atom+xml stream.
+    /// Use this class to materialize objects from a stream.
     /// </summary>
     internal class MaterializeAtom : IDisposable, IEnumerable, IEnumerator
```
Comment (gathogojr) [inline src/Microsoft.OData.Client/MaterializeFromAtom.cs:26]: Maybe change "MaterializeAtom" to "MaterializeObject" given the summary doc and also because "MaterializeEntry" represents something different

File: src/Microsoft.OData.Client/MediaEntryAttribute.cs [hunk 4]
```diff
@@ -11,7 +11,7 @@ namespace Microsoft.OData.Client
     /// <summary>
     /// This class marks a type that represents an Astoria client entity
     /// such that the Astoria client will treat it as a media entry
-    /// according to ATOM's "media link entry" concept.
+    /// according to a"media link entry" concept.
```
Comment (gathogojr) [inline src/Microsoft.OData.Client/MediaEntryAttribute.cs:14]: nit:
```suggestion
    /// according to a "media link entry" concept.
```

File: src/Microsoft.OData.Client/Microsoft.OData.Client.Common.txt [hunk 5]
```diff
@@ -19,7 +19,7 @@ Context_ResolveReturnedInvalidUri=The Uri that is returned by the ResolveEntityS
 Context_RequestUriIsRelativeBaseUriRequired=Because the requestUri is a relative Uri, you must set the BaseUri property on the DataServiceContext.
 Context_ResolveEntitySetOrBaseUriRequired=The ResolveEntitySet function must return a non-null Uri for the EntitySet '{0}', otherwise you must set the BaseUri property.
 Context_CannotConvertKey=Unable to convert value '{0}' into a key string for a URI.
-Context_TrackingExpectsAbsoluteUri=The identity value specified by either the Atom id element or the OData-EntityId header must be an absolute URI.
+Context_TrackingExpectsAbsoluteUri=The identity value specified by either the OData-EntityId header must be an absolute URI.
```
Comment (gathogojr) [inline src/Microsoft.OData.Client/Microsoft.OData.Client.Common.txt:22]:
```suggestion
Context_TrackingExpectsAbsoluteUri=The identity value specified by either the entry id element or OData-EntityId header must be an absolute URI.
```

File: src/Microsoft.OData.Core/Microsoft.OData.Core.csproj [hunk 6]
```diff
@@ -95,4 +95,21 @@
...
+      <AutoGen>True</AutoGen>
+      <DependentUpon>Microsoft.OData.Core.tt</DependentUpon>
+    </Compile>
+    <Compile Update="Parameterized.Microsoft.OData.Core.cs">
+      <DesignTime>True</DesignTime>
+      <AutoGen>True</AutoGen>
+      <DependentUpon>Parameterized.Microsoft.OData.Core.tt</DependentUpon>
+    </Compile>
+  </ItemGroup>
+
```
Comment (gathogojr) [inline src/Microsoft.OData.Core/Microsoft.OData.Core.csproj:114]: Why this change?

File: src/Microsoft.OData.Core/ODataMessageReaderSettings.cs [hunk 7]
```diff
@@ -104,10 +104,6 @@ public ValidationKinds Validations
         /// ODataMessageReaderSettings.BaseUri may be deprecated in the future, please use ODataMessageReaderSettings.baseUri instead.
         /// </summary>
         /// <returns>The base URI used in payload.</returns>
-        /// <remarks>
-        /// This URI will be used in ATOM format only, it would be overridden by &lt;xml:base /&gt; element in ATOM payload.
-        /// If the URI does not end with a slash, a slash would be appended automatically.
```
Comment (gathogojr) [inline src/Microsoft.OData.Core/ODataMessageReaderSettings.cs:109]: Maybe return this line in the remarks

Comment (ElizabethOkerio) [reply src/Microsoft.OData.Core/ODataMessageReaderSettings.cs:109]: Will it still have the atom format explanation?

File: src/Microsoft.OData.Core/Microsoft.OData.Core.csproj (hunk 6 above)
Comment (ElizabethOkerio) [reply src/Microsoft.OData.Core/Microsoft.OData.Core.csproj:114]: I think these got added when I ran the .tt files.

File: src/Microsoft.OData.Core/ODataMessageReaderSettings.cs (hunk 7 above)
Comment (gathogojr) [reply src/Microsoft.OData.Core/ODataMessageReaderSettings.cs:109]: @ElizabethOkerio Not both lines. The second line: "If the URI does not end with a slash, a slash would be appended automatically."
I think it's relevant even for non-ATOM scenarios

File: src/Microsoft.OData.Client/MaterializeFromObject.cs [hunk 8]
```diff
@@ -1,5 +1,5 @@
 ﻿//---------------------------------------------------------------------
-// <copyright file="MaterializeFromAtom.cs" company="Microsoft">
+// <copyright file="MaterializeFromObject.cs" company="Microsoft">
```
Comment (gathogojr) [inline src/Microsoft.OData.Client/MaterializeFromObject.cs:2]: Any reason why the filename doesn't have the same name as the class?

Comment (ElizabethOkerio) [reply src/Microsoft.OData.Client/MaterializeFromObject.cs:2]: Not really. It was like that before. I can change it.

**Review by gathogojr — APPROVED** [review]
